datafusion/functions/src/core/named_struct.rs

```diff
@@ -165,3 +162,73 @@ impl ScalarUDFImpl for NamedStructFunc {
         named_struct_expr(args)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use arrow::array::{BooleanArray, Float64Array, Int64Array};
+    use datafusion_common::cast::as_struct_array;
+    use datafusion_common::ScalarValue;
+
+    #[test]
+    fn test_named_struct() {
+        // struct(1, 2.0, true) = { "foo": 1, "bar": 2.0, "baz": true }
```

Comment:
Most of the test is already covered in `datafusion/sqllogictest/test_files/struct.slt`

Reply:
Okay, should the tests be removed then? I was following the lead of `struct.rs`, so should the test in that file be removed too for consistency?

Reply:
Sure!

Reply:
Here are the instructions: https://github.com/apache/datafusion/tree/main/datafusion/sqllogictest

Ideally you should be able to extend one of the existing test files in https://github.com/apache/datafusion/tree/main/datafusion/sqllogictest/test_files